pymc3/stats.py

```diff
@@ -478,7 +481,7 @@ def gradient(w):
                      bounds=bounds,
                      constraints=constraints)
 
-        weights = w_fuller(w['x'])
+        weights = w_fuller(np.round(w['x'], round_to))
```

Comment:
I have tried those experiment you are suggesting but without luck :-( in fact I end using the two inequalities because for a moment I thought that solved the problem, but then I realize that's not true, then I decide to use the rounding trick. 

I propose something to you, we use the rounding trick for now and in the next few days I will revisit the idea of "tunning" the optimization. What do you think?